### PR TITLE
Add CEED_EVAL_GRAD support for CeedBasisApplyAtPoints

### PR DESCRIPTION
**Pull request by jeremylt**
This adds `CEED_EVAL_GRAD` for `CeedBasisApplyAtPoints` for `CEED_NOTRANSPOSE` mode.

### CODE REVIEW

File: interface/ceed-basis.c [hunk 1]
```diff
@@ -1539,29 +1544,79 @@ int CeedBasisApplyAtPoints(CeedBasis basis, CeedInt num_points, CeedTransposeMod
...
+        case CEED_EVAL_GRAD: {
+          CeedScalar tmp[2][num_comp * CeedIntPow(Q_1d, dim)], chebyshev_x[Q_1d];
+
+          // ---- Values at point
+          for (CeedInt p = 0; p < num_points; p++) {
+            // Dim**2 contractions, apply grad when pass == dim
+            for (CeedInt pass = dim - 1; pass >= 0; pass--) {
+              CeedInt pre = num_comp * CeedIntPow(Q_1d, dim - 1), post = 1;
+
+              for (CeedInt d = dim - 1; d >= 0; d--) {
```
Comment (jeremylt) [inline interface/ceed-basis.c:1582]: Here is this "backwards" silliness again

File: interface/ceed-basis.c [hunk 2]
```diff
@@ -1539,29 +1544,79 @@ int CeedBasisApplyAtPoints(CeedBasis basis, CeedInt num_points, CeedTransposeMod
...
+
+                  chebyshev_temp[1] = 1.0;
+                  chebyshev_temp[2] = 2 * x;
+                  chebyshev_x[0]    = 0.0;
+                  chebyshev_x[1]    = 2.0;
+                  for (CeedInt j = 2; j < Q_1d; j++) {
+                    chebyshev_temp[0] = chebyshev_temp[1];
+                    chebyshev_temp[1] = chebyshev_temp[2];
+                    chebyshev_temp[2] = 2 * x * chebyshev_temp[1] - chebyshev_temp[0];
+                    chebyshev_x[j]    = 2 * x * chebyshev_x[j - 1] + 2 * chebyshev_temp[1] - chebyshev_x[j - 2];
```
Comment (jedbrown) [inline interface/ceed-basis.c:1599]: Might want to put this in a little helper to reduce clutter.

**Review by jedbrown — APPROVED** [review]
Looks good.

File: interface/ceed-basis.c (hunk 1 above)
Comment (jeremylt) [reply interface/ceed-basis.c:1582]: I still don't quite see why it's 'backwards' from the order I expect. I've been pondering on it.

Comment (jeremylt) [reply interface/ceed-basis.c:1618]: Added a note since it works as is

**Comment by jeremylt** [comment]
Should be good to merge now - Nother CI failing was a fluke (that intermittent coverage data overwrite bug)

**Pull request merged**